java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitutionSolver.java

```diff
@@ -86,6 +86,12 @@ JavaType getReturnType(@Nullable JavaType returnType, JavaType defSite, JavaType
         return defSite;
       }
     }
+    // As per getClass javadoc:
+    // The actual result type [of getClass] is Class<? extends |X|> where |X| is the erasure of the static type of the expression on which getClass is called.
+    if(defSite == symbols.objectType && "getClass".equals(method.name())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)